src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -1,138 +1,433 @@
-import { chakra } from "@chakra-ui/react";
-import React, { forwardRef } from "react";
+import React, { useState, forwardRef } from "react";
+import { Box, chakra, useMultiStyleConfig } from "@chakra-ui/react";
 
-import MultiSelectListBox from "./MultiSelectListbox";
-import MultiSelectDialog from "./MultiSelectDialog";
+import Accordion from "./../Accordion/Accordion";
+import CheckboxGroup from "./../CheckboxGroup/CheckboxGroup";
+import Checkbox from "./../Checkbox/Checkbox";
+import MultiSelectMenuButton from "./MultiSelectMenuButton";
+import TextInput from "../TextInput/TextInput";
 
 export interface MultiSelectItem {
   id: string;
   name: string;
+  isDisabled?: boolean;
   children?: MultiSelectItem[];
 }
-
-export const multiSelectWidthsArray = [
-  "default",
-  "fitContent",
-  "full",
-] as const;
+export const multiSelectWidthsArray = ["fitContent", "full"] as const;
 export type MultiSelectWidths = typeof multiSelectWidthsArray[number];
-
+export const multiSelectListOverflowArray = ["scroll", "expand"] as const;
+export type multiSelectListOverflowTypes =
+  typeof multiSelectListOverflowArray[number];
 export interface SelectedItems {
   [name: string]: { items: string[] };
 }
-interface MultiSelectCommonProps {
+type MultiSelectOnChange = (event: React.ChangeEvent<HTMLInputElement>) => void;
+
+export type MultiSelectProps = {
+  /** The helperText sets the text for the internal HelperErrorText */
+  helperText: string;
   /** The id of the MultiSelect. */
   id: string;
   /** Set the default open or closed state of the Multiselect. */
-  isDefaultOpen?: boolean;
+  isDefaultOpen: boolean;
+  /** Boolean value used to enable the component's search functionality
+   * The default value is false. */
+  isSearchable: boolean;
   /** Boolean value used to control how the MultiSelect component will render within the page and interact with other DOM elements.
    * The default value is false. */
-  isBlockElement?: boolean;
+  isBlockElement: boolean;
   /** The items to be rendered in the Multiselect as checkbox options. */
   items: MultiSelectItem[];
+
+  listOverflow?: multiSelectListOverflowTypes;
   /** The label text rendered within the MultiSelect. */
-  labelText: string;
+  buttonText: string;
+  /** Numeric value to set the maximum number of checkbox items
+   * The default value is 5. */
+  defaultItemsVisible: number;
   /** The action to perform for clear/reset button of MultiSelect. */
   onClear?: () => void;
-  /** The type of MultiSelect that will be rendered. */
-  type: "listbox" | "dialog";
+  /** The action to perform on the checkbox's onChange function.  */
+  onChange?: MultiSelectOnChange;
```

Comment:
It seems you're missing to implement `onChangeSearch` and `onClearSearch`. The spec doc says "MVP Req. NO" so I'm assuming it's okay to not implement this now, is that right @bigfishdesign13 ?